src/tests/system/lib/sssd/utils/authentication.py

```diff
@@ -144,6 +171,101 @@ def test_example(client: Client, ldap: LDAP, kdc: KDC):
         return KerberosAuthenticationUtils(self.host, ssh)
 
 
+class PasskeyAuthenticationUtils(MultihostUtility[MultihostHost]):
+    """
+    Methods for testing passkey registration and authentications
+    with umockdev-run command.
+    """
+    def __init__(self, host: MultihostHost, fs: LinuxFileSystem) -> None:
+        """
+        :param host: Multihost host.
+        :type host: MultihostHost
+        :param fs: Linux File system.
+        :type fs: LinuxFileSystem.
+        """
+
+        super().__init__(host)
+        self.fs: LinuxFileSystem = fs
+
+    def register(self, tc: str, cmd: str, pin: str | int) -> SSHProcessResult:
+        """
+        Check umockdev run, while registering the user.
+
+        :param tc: test case variable we use to save the record files.
+        :type tc: str
+        :param cmd: command we use while registering the passkey.
+        :type cmd: str
+        :param pin: pin we set while creating the umockde records.
+        :type pin: str | int
+        :return: True if command runs successful, False otherwise.
+        :rtype: SSHProcessResult
+        """
+
+        # Command we use to run to check register key with umockdev
+
+        command = f"LD_PRELOAD=/opt/random.so umockdev-run --device /tmp/yk.umockdev " \
+                  f"--ioctl /dev/hidraw1=/tmp/{tc}.ioctl " \
+                  f"--script /dev/hidraw1=/tmp/{tc}.script -- {cmd}"
+
+        result = self.host.ssh.expect(rf"""
+                # It takes some time to get authentication failure
+                set timeout 10
```

Comment:
The code timeout if the key isn't connected is 15 so I think you should change this one also.